core/src/main/java/com/nukesz/game/screen/GameScreen.java

```diff
@@ -49,25 +51,27 @@
     private GlyphLayout layout = new GlyphLayout();
     private Texture snakeHead;
     private Texture snakeBody;
-    private Texture apple;
     private Texture wall;
-    private boolean appleAvailable = false;
-    private int appleX;
-    private int appleY;
+    Apple appleObj;
     private Array<BodyPart> bodyParts = new Array<>();
     private int snakeXBeforeUpdate;
     private int snakeYBeforeUpdate;
     private boolean directionSet = false;
     private State state = State.INITIAL;
     private int score = 0;
     private Array<Wall> walls;
-
+    OnscreenControlRenderer controlRenderer;
+    private boolean isSideControllerNeeded =true;
     public GameScreen(SnakeGame game) {
         super(game);
+        if (isSideControllerNeeded) {
+            controlRenderer = new OnscreenControlRenderer();
+        }
     }
 
     @Override
     public void show() {
+        Gdx.graphics.setWindowedMode(WORLD_WIDTH, WORLD_HEIGHT);
```

Comment:
Do we really need this? I have never seen the usage of it.

Reply:
Looks better with it. Please check the game with and without it if you still think to okay to remove I will go with that.